provider/pkg/await/await.go

```diff
@@ -491,9 +499,69 @@ func ssaUpdate(c *UpdateConfig, liveOldObj *unstructured.Unstructured, client dy
 	return currentOutputs, nil
 }
 
+// handleSSAIgnoreFields handles updating the inputs to either drop fields that are present on the cluster and not managed
+// by the current field manager, or to set the value of the field to the last known value applied to the cluster.
+func handleSSAIgnoreFields(c *UpdateConfig, liveOldObj *unstructured.Unstructured) error {
+	managedFields := liveOldObj.GetManagedFields()
+	managedFieldsSet := fieldpath.NewSet()
+
+	for _, f := range managedFields {
+		if f.Manager == c.FieldManager {
+			// We don't need to keep track of fields that we are currently managing.
+			continue
+		}
+		s := fieldpath.NewSet()
+		if err := s.FromJSON(bytes.NewReader(f.FieldsV1.Raw)); err != nil {
+			return fmt.Errorf("unable to parse managed fields from resource %q: %w", c.Inputs.GetName(), err)
+		}
+		managedFieldsSet = managedFieldsSet.Union(s)
+	}
+
+	for _, ignorePath := range c.IgnoreChanges {
+		pathComponents := strings.Split(ignorePath, ".")
+		pe, err := fieldpath.MakePath(makeInterfaceSlice(pathComponents)...)
+		if err != nil {
+			return fmt.Errorf("unable to normalize ignoreField path %q: %w", ignorePath, err)
+		}
+
+		// Drop the field from the inputs if it is present on the cluster and not managed by the current field manager. This ensures
+		// that we don't get any conflict errors, or mistakenly setting the current field manager as a shared manager of that field.
+		if managedFieldsSet.Has(pe) {
+			unstructured.RemoveNestedField(c.Inputs.Object, pathComponents...)
+			continue
+		}
+
+		// We didn't find another field manager that is managing this field, so we need to use the last known value applied to
+		// the cluster so we don't unset it.
+		// NOTE: If the field has been reverted to its default value, ignoreChanges will still not update this field to what is supplied
+		// by the user in their Pulumi program.
+		lastVal, found, err := unstructured.NestedFieldCopy(liveOldObj.Object, pathComponents...)
+		if found && err == nil {
+			err := unstructured.SetNestedField(c.Inputs.Object, lastVal, pathComponents...)
+			if err != nil {
+				return fmt.Errorf("unable to set field %q with last used value %q: %w", ignorePath, lastVal, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func makeInterfaceSlice(inputs []string) []interface{} {
```

Comment:
Nit: This may as well be generic.

```suggestion
func makeInterfaceSlice[T any](inputs []T) []interface{} {
```